components/builder-api/src/server/resources/origins.rs

```diff
@@ -430,60 +411,24 @@ fn create_keys(req: HttpRequest, path: Path<String>, state: Data<AppState>) -> H
-                                             origin:    &origin,
-                                             full_name: &format!("{}-{}", &origin, &pair.rev),
-                                             name:      &origin,
-                                             revision:  &pair.rev,
-                                             body:      &pk_body, };
+    // For Builder, we actually don't want to go through the KeyCache
+    // to create a pair, because we don't want to store anything to
+    // disk. That's why we have a database.
+    let (public, secret) = generate_signing_key_pair(&origin);
 
-    match OriginPublicSigningKey::create(&new_pk, &*conn).map_err(Error::DieselError) {
-        Ok(_) => (),
-        Err(err) => {
-            error!("create_keys: Failed to create public key, err={}", err);
-            return err.into();
-        }
+    if let Err(e) = save_public_origin_signing_key(account_id, &origin, &public, &*conn) {
+        error!("create_keys: Failed to create public key, err={}", e);
+        return e.into();
     }
 
-    let sk_body = match pair.to_secret_string().map_err(Error::HabitatCore) {
-        Ok(sk) => sk.into_bytes(),
-        Err(err) => {
-            error!("create_keys: Failed to get sk body, err={}", err);
-            return err.into();
-        }
-    };
-
-    let (sk_encrypted, bldr_key_rev) = match encrypt(&req, &Bytes::from(sk_body)) {
-        Ok((encrypted, rev)) => (encrypted, rev),
-        Err(err) => {
-            debug!("create_keys: Failed to encrypt sk_body, err={:?}", err);
-            return err.into();
-        }
-    };
-
-    let new_sk = NewOriginPrivateSigningKey { owner_id:           account_id as i64,
-                                              origin:             &origin,
-                                              full_name:          &format!("{}-{}",
-                                                                           &origin, &pair.rev),
-                                              name:               &origin,
-                                              revision:           &pair.rev,
-                                              body:               &sk_encrypted.as_bytes(),
-                                              encryption_key_rev: &bldr_key_rev, };
-
-    match OriginPrivateSigningKey::create(&new_sk, &*conn).map_err(Error::DieselError) {
-        Ok(_) => (),
-        Err(err) => {
-            error!("create_keys: Failed to create private key, err={}", err);
-            return err.into();
-        }
+    if let Err(e) = save_secret_origin_signing_key(account_id,
+                                                   &origin,
+                                                   &state.config.api.key_path,
+                                                   &secret,
+                                                   &*conn)
+    {
+        error!("create_keys: Failed to create private key, err={}", e);
```

Comment:
It might be valuable to capture the origin here. Same for line 420 above. This could prove useful when sifting through the logs. Thoughts on the value add?  Just thinking that if we do end up adding origin to the error, it might also be nice to distinguish that it's the origin private|public _signing_ key.

Reply:
Sounds good!